test/dom/propertyShadowing.ts

```diff
@@ -22,6 +28,29 @@ describe('dom.Struct property shadowing', () => {
         assert.equal(s.get('fieldNames', 1)!.stringValue(), 'cat');
         assert.equal(s.get('fieldNames', 2)!.stringValue(), 'mouse');
 
+        // deleteProperty proxy method
+        assert.isTrue(delete s['greetings']);
+        assert.deepEqual(s.fieldNames(), ['getType', 'getAnnotations', 'fieldNames', 'toString', 'age']);
+
+        // deleteField Struct method
+        assert.isTrue(s.deleteField('age'));
+        assert.deepEqual(s.fieldNames(), ['getType', 'getAnnotations', 'fieldNames', 'toString']);
+
+        // delete for properties that match built-in
+        assert.equal(s.get('toString')!.numberValue(), 10);
+        assert.isTrue(delete s['toString']);
+        assert.equal(s.toString(),'{getType: baz, getAnnotations: 56, fieldNames: [dog, cat, mouse]}');
```

Comment:
This test might be somewhat fragile; I don't know how rigidly the behavior of `toString()` is defined by the ECMA spec. If a web browser or NodeJS decided to add newlines to the returned string, for example, this would break.

We can achieve the same thing by just making sure that `toString` is still present and is a function:
```suggestion
        assert.equal(typeof s.toString, "function");
```